### ISSUE TITLE
Don't see subscription yearly interval

### ISSUE THREAD

**Issue by noecom12** [nextcloud/calendar#2068]
### Steps to reproduce
1. add New subscription from link (read-only) from a Microsoft Exchange Server 2010

https://blablabla.com/calendar.ics
> BEGIN:VCALENDAR
> METHOD:PUBLISH
> PRODID:Microsoft Exchange Server 2010
> VERSION:2.0
> X-WR-CALNAME:BDay
> BEGIN:VTIMEZONE
> TZID:Central Europe Standard Time
> BEGIN:STANDARD
> DTSTART:16010101T030000
> TZOFFSETFROM:+0200
> TZOFFSETTO:+0100
> RRULE:FREQ=YEARLY;INTERVAL=1;BYDAY=-1SU;BYMONTH=10
> END:STANDARD
> BEGIN:DAYLIGHT
> DTSTART:16010101T020000
> TZOFFSETFROM:+0100
> TZOFFSETTO:+0200
> RRULE:FREQ=YEARLY;INTERVAL=1;BYDAY=-1SU;BYMONTH=3
> END:DAYLIGHT
> END:VTIMEZONE
> BEGIN:VEVENT
> RRULE:FREQ=YEARLY;UNTIL=20200221T230000Z;INTERVAL=1;BYMONTHDAY=22;BYMONTH=2
> UID:040000008200E00074C5B7101A82E00800000000409E8A6E9FE8D501000000000000000
>  01000000087A1CC656CFE204BA46194F1DB546E90
> SUMMARY:Karel BDay
> DTSTART;VALUE=DATE:20200222
> DTEND;VALUE=DATE:20200223
> CLASS:PUBLIC
> PRIORITY:5
> DTSTAMP:20200310T194004Z
> TRANSP:TRANSPARENT
> STATUS:CONFIRMED
> SEQUENCE:0
> LOCATION:
> X-MICROSOFT-CDO-APPT-SEQUENCE:0
> X-MICROSOFT-CDO-BUSYSTATUS:FREE
> X-MICROSOFT-CDO-INTENDEDSTATUS:BUSY
> X-MICROSOFT-CDO-ALLDAYEVENT:TRUE
> X-MICROSOFT-CDO-IMPORTANCE:1
> X-MICROSOFT-CDO-INSTTYPE:1
> X-MICROSOFT-DONOTFORWARDMEETING:FALSE
> X-MICROSOFT-DISALLOW-COUNTER:FALSE
> END:VEVENT
> BEGIN:VEVENT
> RRULE:FREQ=YEARLY;UNTIL=20200310T230000Z;INTERVAL=1;BYMONTHDAY=11;BYMONTH=3
> UID:040000008200E00074C5B7101A82E0080000000090081821DCE8D501000000000000000
>  010000000CD756389AEEBED4781BE6748C5F48203
> SUMMARY:Piet BDay
> DTSTART;VALUE=DATE:20200411
> DTEND;VALUE=DATE:20200412
> CLASS:PUBLIC
> PRIORITY:5
> DTSTAMP:20200310T194004Z
> TRANSP:TRANSPARENT
> STATUS:CONFIRMED
> SEQUENCE:0
> LOCATION:
> X-MICROSOFT-CDO-APPT-SEQUENCE:0
> X-MICROSOFT-CDO-BUSYSTATUS:FREE
> X-MICROSOFT-CDO-INTENDEDSTATUS:BUSY
> X-MICROSOFT-CDO-ALLDAYEVENT:TRUE
> X-MICROSOFT-CDO-IMPORTANCE:1
> X-MICROSOFT-CDO-INSTTYPE:1
> X-MICROSOFT-DONOTFORWARDMEETING:FALSE
> X-MICROSOFT-DISALLOW-COUNTER:FALSE
> END:VEVENT
> END:VCALENDAR

2. Look in the calendar, and you don't see the yearly intervals.
3.

### Expected behaviour
Tell us what should happen
You see the yearly intervals

### Actual behaviour
Tell us what happens instead
You dont see the yearly intervals

### Calendar app
**Calendar app version:** (see apps admin page, e.g. 2.0.1)
2.0.2

**CalDAV-clients used:** (Thunderbird Lightning, DAVx5, Evolution, macOS Calendar, etc)
Browser Chrome

### Client configuration
**Browser:** (e.g. Firefox 48)
Browser Chrome

**Operating system:** (e.g. Arch Linux)
Windows 10

### Server configuration

**Operating system**: (e.g. Debian 8)
Debian 10

**Web server:** (e.g. Apache, Nginx,...)
Apache

**Database:** (e.g. MariaDB, SQLite or PostgreSQL)
MariaDB

**PHP version:** (e.g. 7.0.3)
PHP 7.3

**Nextcloud Version:** (see admin page, e.g. 17.0.2)
18.0.1

**Updated from an older installed version or fresh install:**
Updated from an older installed version

**List of activated apps:**

```
If you have access to your command line run e.g.:
sudo -u www-data php occ app:list
from within your instance's installation folder
```

  - accessibility: 1.4.0
  - activity: 2.11.0
  - calendar: 2.0.2
  - checksum: 0.4.4
  - cloud_federation_api: 1.1.0
  - contacts: 3.2.0
  - dav: 1.14.0
  - deck: 0.8.0
  - emlviewer: 0.0.5
  - event_update_notification: 1.0.1
  - external: 3.5.0
  - federatedfilesharing: 1.8.0
  - federation: 1.8.0
  - files: 1.13.1
  - files_downloadactivity: 1.7.0
  - files_external: 1.9.0
  - files_markdown: 2.2.0
  - files_pdfviewer: 1.7.0
  - files_rightclick: 0.15.2
  - files_sharing: 1.10.1
  - files_trashbin: 1.8.0
  - files_versions: 1.11.0
  - files_videoplayer: 1.7.0
  - logreader: 2.3.0
  - lookup_server_connector: 1.6.0
  - mail: 1.1.3
  - metadata: 0.11.1
  - nextcloud_announcements: 1.7.0
  - notifications: 2.6.0
  - oauth2: 1.6.0
  - password_policy: 1.8.0
  - passwords: 2020.3.1
  - photos: 1.0.0
  - privacy: 1.2.0
  - provisioning_api: 1.8.0
  - serverinfo: 1.8.0
  - settings: 1.0.0
  - survey_client: 1.6.0
  - systemtags: 1.8.0
  - text: 2.0.0
  - theming: 1.9.0
  - theming_customcss: 1.5.0
  - twofactor_backupcodes: 1.7.0
  - twofactor_totp: 4.1.2
  - updatenotification: 1.8.0
  - viewer: 1.2.0
  - workflowengine: 2.0.0
Disabled:
  - admin_audit
  - comments
  - encryption
  - firstrunwizard
  - recommendations
  - sharebymail
  - support
  - user_ldap

**Nextcloud configuration:**

```
If you have access to your command line run e.g.:
sudo -u www-data php occ config:list system
from within your instance's installation folder

or

Insert your config.php content here
Make sure to remove all sensitive content such as passwords. (e.g. database password, passwordsalt, secret, smtp password, …)
```

Null

#### Web server error log (e.g. /var/log/apache)
```
Insert your webserver log here
```
Null

#### Log file (data/nextcloud.log)
```
Insert your nextcloud.log file here
```
Null

#### Browser log
```
Insert your browser log here, this could for example include:

a) The javascript console log
b) The network log
c) ...
```
Null

**Comment by georgehrke** [comment]
This has been fixed in the server: nextcloud/server#19398

Fix will be shipped with 18.0.2

**Comment by georgehrke** [comment]
Looking into the ICS Data, it's also an invalid ICS file:

```
...
RRULE:FREQ=YEARLY;UNTIL=20200221T230000Z;INTERVAL=1;BYMONTHDAY=22;BYMONTH=2
...
DTSTART;VALUE=DATE:20200222
...
```

The event starts on 22nd Feb 2020, but the recurrence already ends on 21st Feb 2020.
=> If the recurrence-set ends before it even starts, we don't display any events.
